tutor/commands/jobs.py

```diff
@@ -0,0 +1,236 @@
+"""
+Common jobs that must be added both to local, dev and k8s commands.
+"""
+import functools
+import typing as t
+
+import click
+from typing_extensions import ParamSpec
+
+from tutor import config as tutor_config
+from tutor import env, fmt, hooks
+
+
+def add_job_commands(do_command_group: click.Group) -> None:
+    """
+    This is meant to be called with the `local/dev/k8s do` group commands, to add the
+    different `do` subcommands.
+    """
+    subcommands: t.Iterator[click.Command] = hooks.Filters.CLI_DO_COMMANDS.iterate()
+    for subcommand in subcommands:
+        do_command_group.add_command(subcommand)
+
+
+@hooks.Actions.CORE_READY.add()
+def _add_core_init_tasks() -> None:
+    """
+    Declare core init scripts at runtime.
+
+    The context is important, because it allows us to select the init scripts based on
+    the --limit argument.
+    """
+    with hooks.Contexts.APP("mysql").enter():
+        hooks.Filters.COMMANDS_INIT.add_item(("mysql", ("hooks", "mysql", "init")))
+    with hooks.Contexts.APP("lms").enter():
+        hooks.Filters.COMMANDS_INIT.add_item(("lms", ("hooks", "lms", "init")))
+    with hooks.Contexts.APP("cms").enter():
+        hooks.Filters.COMMANDS_INIT.add_item(("cms", ("hooks", "cms", "init")))
```

Comment:
Since all the updates we've made to the Plugins API, the name and structure of the `hooks/` template directory has stopped making sense.

What about, instead, having a single, flat `init/` directory?

```suggestion
        hooks.Filters.COMMANDS_INIT.add_item(("mysql", ("init", "mysql")))
    with hooks.Contexts.APP("lms").enter():
        hooks.Filters.COMMANDS_INIT.add_item(("lms", ("init", "lms")))
    with hooks.Contexts.APP("cms").enter():
        hooks.Filters.COMMANDS_INIT.add_item(("cms", ("init", "cms")))
```
(I'd also support holding off on these sort of refactorings until the big tutorcore extraction.)

Reply:
I think it's fine if we move the init scripts. It should not be a breaking change. I'll move the scripts to `tutor/templates/jobs/init/<service>.sh`.